src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -0,0 +1,322 @@
+        }
+
+        /// <summary>
+        /// Called just prior to launch to do additionl work (put up ui, do special configuration etc).
+        /// </summary>
+        public Task OnBeforeLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile)
+        {
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Called just after the launch to do additionl work (put up ui, do special configuration etc).
+        /// </summary>
+        public Task OnAfterLaunchAsync(DebugLaunchOptions launchOptions, ILaunchProfile profile)
+        {
+            return Task.CompletedTask;
+        }
+
+        private async Task<bool> GetIsClassLibraryAsync()
+        {
+            // Used by default Windows debugger to figure out whether to add an extra
+            // pause to end of window when CTRL+F5'ing a console application
+            var configuration = await Properties.GetConfigurationGeneralPropertiesAsync()
+                                                 .ConfigureAwait(false);
+
+
+            IEnumValue outputType = (IEnumValue)await configuration.OutputType.GetValueAsync()
+                                                                      .ConfigureAwait(false);
+
+            return StringComparers.PropertyValues.Equals(outputType.Name, ConfigurationGeneral.OutputTypeValues.Library);
+        }
+
+        /// <summary>
+        /// This is called on F5\Ctrl-F5 to return the list of debug targets.What we return depends on the type
+        /// of project.  The dth port contains the port of the design time host. The cmdArgs will only be 
+        /// specified if a command (from project.json) was selected as the target. Note that debugging
+        /// url can be null
+        /// </summary>
+        public async Task<IReadOnlyList<IDebugLaunchSettings>> QueryDebugTargetsAsync(DebugLaunchOptions launchOptions, ILaunchProfile activeProfile)
+        {
+
+            string projectFolder = Path.GetDirectoryName(ConfiguredProject.UnconfiguredProject.FullPath);
+
+            List<DebugLaunchSettings> launchTargets = new List<DebugLaunchSettings>();
+
+            var settings = new DebugLaunchSettings(launchOptions);
+
+            // Resolve the tokens in the profile
+            ILaunchProfile resolvedProfile = await TokenReplacer.ReplaceTokensInProfileAsync(activeProfile).ConfigureAwait(true);
+
+            // We want to launch the process via the command shell when not debugging, except when this debug session is being launched for profiling.
+            bool useCmdShell = (launchOptions & (DebugLaunchOptions.NoDebug | DebugLaunchOptions.Profiling)) == DebugLaunchOptions.NoDebug;
+            var conosoleTarget = await GetConsoleTargetForProfile(resolvedProfile, launchOptions, projectFolder, useCmdShell).ConfigureAwait(true);
```

Comment:
`var` is usually a good candidate when the return type is well known or could be inferred from the variable name. So I would recommend using the actual type name, `DebugLaunchSettings` or rename the variable to `consoleTargetDebugLaunchSettings`. Long variable names are fine as long as they are self explanatory